src/libraries/Microsoft.Extensions.Options/gen/Model/ValidatedModel.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Options.Generators
+{
+    internal sealed record class ValidatedModel(
+        string Name,
+        string SimpleName,
+        bool SelfValidates,
+        List<ValidatedMember> MembersToValidate);
```

Comment:
Instead of List, consider using a collection type with sequence equality so that your model has the desired equality semantics. Here's one from the repo that you can use:

https://github.com/dotnet/runtime/blob/be6a1389c852fdc96a919769867fb3f3ddb0fbde/src/libraries/System.Text.Json/gen/Helpers/ImmutableEquatableArray.cs

You could try moving the class to the common folder so that it's shared between the two projects.